### DISCUSSION TITLE
Filter questions

### DISCUSSION THREAD

**Discussion by jedwards4b** [Unidata/netcdf-c#2648] [Q&A]
We have written a new filter which can be run on the GPU.   My question is that if we use the new filter to compress using deflate or zstd on GPU is there a mechanism or can we create a mechanism to first try the GPU based filter and if it is not found fall back to the cpu filter to decompress the file?

**Comment by DennisHeimbigner** [comment]
Short answer: no. I think the best you can do is build a single combined filter that chooses the filter to use
based on whatever criteria it uses.

**Reply by jedwards4b** [reply]
So we should add something into our filter that calls back to the cpu zstd or deflate if the gpu were not available.   That makes sense, thank you.
